src/Internal/Payment/FailedTransactionRateLimiter.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Class FailedTransactionRateLimiter
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment;
+
+use WCPay\Internal\Proxy\LegacyProxy;
+use WCPay\Internal\Service\SessionService;
+
+/**
+ * A wrapper class for keeping track of events in registries, and to trigger a rate limiter after a threshold.
+ */
+class FailedTransactionRateLimiter extends AbstractSessionRateLimiter {
```

Comment:
I'd like to receive the feedback on this inheritance. To be honest, I myself don't like this approach but given that, `SessionRateLimiter` (or in the current PR, it's `AbstractSessionRateLimiter`) depends on `SessionService` and `LegacyProxy`, this approach is allowed to use DI nicely with `PaymentsServiceProvider`. Otherwise, it will be something like this: 

https://github.com/Automattic/woocommerce-payments/pull/7673/files#diff-bd48d09e9d1c96e424dd2cf820a10f9aba6b0b2c9752bf7e9e9fb82c33a69889R74-R85

Reply:
Resolved, see https://github.com/Automattic/woocommerce-payments/pull/7672#issuecomment-1809921599